### PR TITLE
Fix for FlexLayout padding issue

### LINKED ISSUE #15991: FlexLayout Padding not working
### Description

Padding does not work in FlexLayout.
It works in Xamarin.Forms and was displayed correctly before migration to MAUI.
After MAUI migration, the layout is broken.

```xml
<VerticalStackLayout BackgroundColor="AliceBlue">
    <FlexLayout Direction="Column" Padding="16" BackgroundColor="LightGreen">
        <BoxView WidthRequest="40" HeightRequest="20" Color="Red"  />
        <BoxView WidthRequest="40" HeightRequest="20" Color="Blue" />
        <BoxView WidthRequest="40" HeightRequest="20" Color="Green"  />
        <BoxView WidthRequest="40" HeightRequest="20" Color="Yellow"/>
        <BoxView WidthRequest="40" HeightRequest="20" Color="Magenta"  />
        <BoxView WidthRequest="40" HeightRequest="20" Color="Black"  />
        <BoxView WidthRequest="40" HeightRequest="20" Color="Orange"  />
        <BoxView WidthRequest="40" HeightRequest="20" Color="GreenYellow"  />
    </FlexLayout>
</VerticalStackLayout>
```

 

### Steps to Reproduce

1. Put a FlexLayout with Padding.

### Link to public reproduction project repository

https://github.com/muak/FlexIssue

### Version with bug

7.0.86

### Last version that worked well

Unknown/Other

### Affected platforms

iOS, Android

### Affected platform versions

iOS15.4 / Android API33

### Did you find any workaround?

Set parent layout padding.

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by BagavathiPerumal**
### Root cause

The FlexLayout doesn't include the padding value during its arrangement and measurement processes. As a result, the padding is not applied in the FlexLayout.

### Description of Issue Fix

The fix involves updating the layout arrangement to include padding values, adjust bounds, and arrange children accordingly. The measurement process is adjusted to exclude padding during measurement, then add it back to determine the final size. This ensures accurate layout and measurement with proper padding.

### Issues Fixed

Fixes #15991

### Reference:

Referred code changes from HorizontalStackLayoutManager.cs file. 

https://github.com/dotnet/maui/blob/210bc3581a8033cbbcbb14da931833c68c85583e/src/Core/src/Layouts/HorizontalStackLayoutManager.cs#L12

### Output

Before fix:

image

After fix:

image

### CODE REVIEW

**Review by PureWeen — CHANGES_REQUESTED** [review]
For this one, instead of doing a screenshot test can we move these tests to the "UnitTests" ? 

Since none of the changes in here depend on anything platformspecific?

Here's an example of some tests against the new FlexLayout

https://github.com/dotnet/maui/blob/c1566589d1794e5adcbcf2e93bd77fc85f859253/src/Controls/tests/Core.UnitTests/Layouts/FlexLayoutTests.cs#L11

**Comment by BagavathiPerumal** [comment]
> For this one, instead of doing a screenshot test can we move these tests to the "UnitTests" ?
> 
> Since none of the changes in here depend on anything platformspecific?
> 
> Here's an example of some tests against the new FlexLayout
> 
> https://github.com/dotnet/maui/blob/c1566589d1794e5adcbcf2e93bd77fc85f859253/src/Controls/tests/Core.UnitTests/Layouts/FlexLayoutTests.cs#L11

**Comment by BagavathiPerumal** [comment]
> For this one, instead of doing a screenshot test can we move these tests to the "UnitTests" ?
> 
> Since none of the changes in here depend on anything platformspecific?
> 
> Here's an example of some tests against the new FlexLayout
> 
> https://github.com/dotnet/maui/blob/c1566589d1794e5adcbcf2e93bd77fc85f859253/src/Controls/tests/Core.UnitTests/Layouts/FlexLayoutTests.cs#L11

@PureWeen I have updated the tests to Unit Tests, could you please validate once and let me know if there is any concerns

File: src/Controls/tests/Core.UnitTests/Layouts/FlexLayoutTests.cs [hunk 1]
```diff
@@ -133,6 +141,34 @@ public void UnconstrainedWidthChildrenHaveWidth()
...
+				HeightRequest = 500,
+				WidthRequest = 500,
+				Children = { boxView }
+			};
+			root.Add(layout);
+			var measure = layout.CrossPlatformMeasure(500, 500);
+			layout.CrossPlatformArrange(new Rect(Point.Zero, measure));
+
+			var frame = layout.Children[0].Frame;
+			Assert.Equal(padding, frame.X);
```
Comment (PureWeen) [inline src/Controls/tests/Core.UnitTests/Layouts/FlexLayoutTests.cs:169]: This is looking good!

Can we check the padding all around the child? Make sure that it's inset correctly from all sides and has the expected width/height?

**Review by PureWeen — CHANGES_REQUESTED** [review]

File: src/Controls/tests/Core.UnitTests/Layouts/FlexLayoutTests.cs (hunk 1 above)
Comment (BagavathiPerumal) [reply src/Controls/tests/Core.UnitTests/Layouts/FlexLayoutTests.cs]: @PureWeen, I have updated the Unit cases based on your concerns, could you please validate once and let me know if there are any concerns.

File: src/Controls/tests/Core.UnitTests/Layouts/FlexLayoutTests.cs [hunk 2]
```diff
@@ -133,6 +133,64 @@ public void UnconstrainedWidthChildrenHaveWidth()
...
+
+			var leftPadding = view1Frame.X;
+			var topPadding = view1Frame.Y;
+			var rightPadding = measure.Width - view2Frame.Right;
+
+			// Assert
+			Assert.Equal(padding, leftPadding);
+			Assert.Equal(padding, rightPadding);
+			Assert.Equal(padding, topPadding);
+			Assert.Equal(view1.Width, view1Frame.Width);
```
Comment (PureWeen) [inline src/Controls/tests/Core.UnitTests/Layouts/FlexLayoutTests.cs:165]: Can this check use a more specific value? 
I think this comparison isn't comparing anything interesting

Check for the specific values you expect the width of view1 and view2 to be

File: src/Controls/tests/Core.UnitTests/Layouts/FlexLayoutTests.cs [hunk 3]
```diff
@@ -133,6 +133,64 @@ public void UnconstrainedWidthChildrenHaveWidth()
...
+			// Act
+			var measure = flexLayout.CrossPlatformMeasure(1000, 1000);
+			flexLayout.CrossPlatformArrange(new Rect(Point.Zero, measure));
+
+			var view2Frame = flexLayout.Children[1].Frame;
+			var bottomPadding = measure.Height - view2Frame.Bottom;
+
+			// Assert
+			Assert.Equal(padding, bottomPadding);
+			Assert.Equal(view2.Height, view2Frame.Height);
```
Comment (PureWeen) [inline src/Controls/tests/Core.UnitTests/Layouts/FlexLayoutTests.cs:191]: Same here

**Review by PureWeen — CHANGES_REQUESTED** [review]

**Review by PureWeen — APPROVED** [review]

**Review by rmarinho — APPROVED** [review]
Failing test not related

**Pull request merged**